editor_canvas.py

```
@@ -75,60 +96,168 @@ def redraw(self):
                 return uml_class
         return None
 
+    def find_relationship_handle_at(self, x, y) -> Tuple[UMLRelationship, HandleType] | None:
+        if not self.selected_relationship:
+            return None
+            
+        rel = self.selected_relationship
+        p0, p1, p2, p3 = self._calculate_bezier_points(rel)
+        
+        threshold = 6
+        if geometry.distance((x, y), p0) <= threshold: return (rel, HandleType.SOURCE_CONNECT)
```

Ruff (0.15.6)

[error] Multiple statements on one line (colon)

(E701)